.github/workflows/doc.yml

```diff
@@ -32,6 +33,38 @@ jobs:
     - uses: actions/checkout@v3
     - name: Register Sphinx problem matcher
       run: echo "::add-matcher::.github/problem-matchers/sphinx.json"
+    - name: 'Set up Python'
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3'
+        cache: 'pip'
+        cache-dependency-path: 'Doc/requirements.txt'
+    - name: 'Install build dependencies'
+      run: make -C Doc/ venv
+    - name: 'Check documentation'
+      run: make -C Doc/ SPHINXOPTS="-q -W --keep-going" check
+    - name: 'Build HTML documentation'
+      run: make -C Doc/ SPHINXOPTS="-q -W --keep-going" html
+    - name: 'Upload'
+      uses: actions/upload-artifact@v3
+      with:
+        name: doc-html
+        path: Doc/build/html
+
+  # Run "doctest" on HEAD as new syntax doesn't exist in the latest stable release
+  doctest:
+    name: 'Doctest'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Register Sphinx problem matcher
+      run: echo "::add-matcher::.github/problem-matchers/sphinx.json"
```

Comment:
How does this lone `echo` work?

Reply:
Again, copy-pasting! It is actions syntax for the problem matchers, the 'echo' is caught by GHA's log interpreter.

A

Reply:
Docs: https://github.com/actions/toolkit/blob/master/docs/commands.md#problem-matchers

Added in https://github.com/python/cpython/pull/20325

Discussed with screenshots in https://discuss.python.org/t/using-github-problem-matchers-to-catch-warnings-early/4254

Reply:
Neat!
Thanks for sharing these links!